Project2015To2017/ProjectConverter.cs

```diff
@@ -37,26 +37,34 @@ public sealed class ProjectConverter
 			this.projectReader = new ProjectReader(logger, this.conversionOptions);
 		}
 
+		[Obsolete("This method is deprecated, call Process[Project|Solution]File directly instead")]
```

Comment:
Is there an easy way I can use the globbing or folder-wide functionality when using this as a package? Before I could pass in a folder to this for it to process all our solutions which is handy in our setup

Reply:
That was intended to be part of user-oriented Migrate2017.Tool functionality. If you control callsite (use this as a NuGet), you can handle things much better on your side rather than rely on library's built-in solutions. Library should provide solid and generic API, using which anyone can build their custom logic.

Reply:
I only want a tiny bit of customisation though (the custom transforms before and after). That's why I added it to this project. It would be annoying to have to roll my own directory searching when the code is all there in the project. Remember this conversion is a one time activity so I don't want to spend ages writing the code to do it.

Reply:
I mean, is this that difficult (the following is written by memory without IDE)?
```cs
var converter = new ProjectConverter(...);
var projects = new List<Project>();
foreach (var x in Directory.EnumerateFiles(Directory.GetCurrentDirectory(), "*.csproj", SearchOption.AllDirectories))
    projects.Add(converter.ProcessProjectFile(new FileInfo(x), null));
```

Reply:
But this piece of code will certainly do only what _you_ need, it will not try doing other things that are not needed in your scenario.

Reply:
Remember I am one of the few users of the package and you are taking a feature away from me while simultaneously implementing it in a bit of the code that I can't make easy use of. I don't work with files and folders as much as you might so no I couldn't have written the above from memory and I would probably get it wrong a couple of times before I got it right.

If we could just add `ProcessFolder`, with a globbing option (maybe even defaulted to `**/*.sln`, then I'll be happy

Reply:
But that would force library to depend on additional NuGet package. If @hvanbakel is fine with that - why not. But that would tempt to use that when better solutions are present (like the one above). Globbing is really expensive, in CLI I first fetch all possible matching files (once in a process lifetime) and only then apply glob matching.

Reply:
I don't think there's an issue with having methods in the core library that we don't use in the cli. They are 2 different interfaces to the same functionality, they don't have to be the same.

I'll do a full review later today

Reply:
I would just leave the `Obsolete` out it doesn't matter that we don't use it, it is useful for the nupkg consumers.